int64 idx
int64 current_idx
geometry_msgs/Pose pose_global
geometry_msgs/Pose pose_surface
float64 ratio
float64 q_weight
string state